storage/series.go

```diff
@@ -23,6 +23,47 @@ import (
 	"github.com/prometheus/prometheus/tsdb/tsdbutil"
 )
 
+type SeriesEntry struct {
```

Comment:
Can we add some comments for `type xxxEntry`? Not very clear just from the name, like what interface is it supposed to implement etc. Maybe the `Entry` suffix is confusing.